components/patient-data/impl/src/main/java/org/phenotips/data/internal/SecurePatientEntityManager.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see http://www.gnu.org/licenses/
+ */
+package org.phenotips.data.internal;
+
+import org.phenotips.entities.PrimaryEntityManager;
+
+import org.xwiki.component.annotation.Component;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+/**
+ * Secure implementation of patient data access service using XWiki as the storage backend.
+ *
+ * @version $Id$
+ * @since 1.4
+ */
+@Named("Patient/secure")
+@Component(roles = {PrimaryEntityManager.class})
+@Singleton
+public class SecurePatientEntityManager extends PatientEntityManager
```

Comment:
This is not secured in any way. You need to also move the relevant code from `SecurePatientRepository` here.

Reply:
Fixed